[sanitizers] Explainer about dyld and weak overrides on Darwin. (NFC)

rsundahl opened the pull request:
Explain in the release notes that the Darwin dynamic linker (dyld) requires that at least one weak symbol be present in any mach-o file that defines an intended override of a sanitizer dylib weak reference.

rdar://103453678

Reviewed By: thetruestblue

Differential Revision: https://reviews.llvm.org/D146745

(cherry picked from commit f1c5c84ae1a87c6ea707ae3b3e524e9eeaf1d70a)

Commit:
[sanitizers] Explainer about dyld and weak overrides on Darwin. (NFC)

Explain in the release notes that the Darwin dynamic linker (dyld) requires
that at least one weak symbol be present in any mach-o file that defines an
intended override of a sanitizer dylib weak reference.

rdar://103453678

Reviewed By: thetruestblue

Differential Revision: https://reviews.llvm.org/D146745

(cherry picked from commit f1c5c84ae1a87c6ea707ae3b3e524e9eeaf1d70a)

yln commented:
I noticed that the open source commit didn't incorporate my feedback: 
https://reviews.llvm.org/D146745#inline-1418668
https://reviews.llvm.org/D146745#inline-1418676

Commit:
[sanitizers] Simplify Explainer about dyld and weak overrides on Darwin. (NFC)

Presenting more than one way to satisfy the single-weak-ref requirement leads to
confusing messaging for the end user. Use the introduction of a single unused
weak variable as the preferred solution. This differential modifies D146745.

rdar://103453678

Reviewed By: yln, thetruestblue

Differential Revision: https://reviews.llvm.org/D147526

(cherry picked from commit 5f17ba1a3832450eee9c3b85ab862f73712184a4)

AaronBallman committed:
Fix LLVM sphinx build

This addresses issues found by:
https://lab.llvm.org/buildbot/#/builders/30/builds/33698

(cherry picked from commit ceacb3252416b450cb42a951eb531b6d66a89da8)

rsundahl commented:
Reviewers, please take a quick look again after consolidation. Thank you.

yln commented:
@swift-ci Please smoke test

yln commented:
@swift-ci Please test
